blindsign/blindrsa/blindrsa.go

```diff
@@ -0,0 +1,213 @@
+package blindrsa
+
+// This package implements the blind RSA protocol based on the CFRG specification:
+// https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-rsa-blind-signatures-02
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/subtle"
+	"errors"
+	"hash"
+	"io"
+	"math/big"
+
+	"github.com/cloudflare/circl/blindsign"
+)
+
+// An RSAVerifier represents a Verifier in the RSA blind signature protocol.
+// It carries state needed to produce and validate an RSA blind signature.
+type RSAVerifier struct {
+	// Public key of the Signer
+	pk *rsa.PublicKey
+
+	// Hash function used in producing the message signature
+	hash hash.Hash
+}
+
+// NewRSAVerifier creates a new RSAVerifier using the corresponding Signer parameters.
+func NewRSAVerifier(pk *rsa.PublicKey, hash hash.Hash) RSAVerifier {
+	return RSAVerifier{
+		pk:   pk,
+		hash: hash,
+	}
+}
+
+func encodeMessageEMSAPSS(message []byte, key *rsa.PublicKey, hash hash.Hash, salt []byte) ([]byte, error) {
+	hash.Write(message)
+	digest := hash.Sum(nil)
+	hash.Reset()
+	emBits := key.N.BitLen() - 1
+	encodedMsg, err := emsaPSSEncode(digest[:], emBits, salt, hash)
+	return encodedMsg, err
+}
+
+func blind(random io.Reader, key *rsa.PublicKey) (*big.Int, *big.Int, error) {
```

Comment:
I was confused by the purpose of this method because it uses "blind" as a noun, whereas the API call "Blind()" uses it as a verb. How about "generateBlindingFactor"?